sapi/phpdbg/phpdbg_prompt.c

```diff
@@ -1802,16 +1802,13 @@ void phpdbg_execute_ex(zend_execute_data *execute_data) /* {{{ */
 			goto next;
 		}
 
-		/* not while in conditionals */
-		phpdbg_print_opline_ex(execute_data, 0);
-
 		/* perform seek operation */
 		if ((PHPDBG_G(flags) & PHPDBG_SEEK_MASK) && !(PHPDBG_G(flags) & PHPDBG_IN_EVAL)) {
 			/* current address */
 			zend_ulong address = (zend_ulong) execute_data->opline;
 
 			if (PHPDBG_G(seek_ex) != execute_data) {
-				if (PHPDBG_G(flags) & PHPDBG_IS_STEPPING) {
+				if (PHPDBG_G(seek_ex)->prev_execute_data == execute_data && PHPDBG_G(flags) & PHPDBG_IS_STEPPING) {
```

Comment:
I think that needs to be `phpdbg_user_execute_data(PHPDBG_G(seek_ex)->prev_execute_data)`

Reply:
Any idea how to test this? I've tried swapping and the current tests work either way, but it's not clear (i'm unfamiliar with the codebase) what at the source level triggers these conditions.

Reply:
Yeah, this obviously fails when called on top of the stack. My bad. Needs to be some check `PHPDBG_G(seek_ex)->prev_execute_data && phpdbg_user_execute_data(PHPDBG_G(seek_ex)->prev_execute_data) == execute_data` I suppose.